File: Branch-SDK/NativeComputeDebugButtonCell.m
//
//  NativeComputeDebugButtonCell.m
//  Branch
//
//  Created by Usman Shafique on 8/26/22.
//  Copyright © 2022 Branch, Inc. All rights reserved.
//

#import "NativeComputeDebugButtonCell.h"

@interface NativeComputeDebugButtonCell()
@property (nonatomic, weak) UIButton *processConfigButton;
@end

@implementation NativeComputeDebugButtonCell

- (id)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    [self initialize];
    return self;
}

- (id)initWithCoder:(NSCoder *)coder {
    self = [super initWithCoder:coder];
    [self initialize];
    return self;
}

- (void)initialize {
    
    // textView config
    UITextView *textView = [[UITextView alloc] initWithFrame:CGRectZero];
    [textView setTranslatesAutoresizingMaskIntoConstraints:NO];
    [textView setEditable:NO];
    [textView setScrollEnabled:NO];
    [textView setEditable:NO];
    [textView setTextAlignment:NSTextAlignmentLeft];
    [textView setBackgroundColor:UIColor.lightGrayColor];
    [textView setTextColor:UIColor.blackColor];
    [self.contentView addSubview:textView];
    self.textView = textView;
    
    // button config
    UIButton *button = [UIButton buttonWithType:UIButtonTypeRoundedRect];
    [button setTranslatesAutoresizingMaskIntoConstraints:NO];
    [button setTitle:@"Process All Config URLs" forState:UIControlStateNormal];
    [button setBackgroundColor:UIColor.blackColor];
    [button setTitleColor:UIColor.lightGrayColor forState:UIControlStateNormal];
    [button.titleLabel setFont:[UIFont systemFontOfSize:14]];
    [button setEnabled:YES];
    [button setUserInteractionEnabled:YES];
    [self.contentView addSubview:button];
    self.processConfigButton = button;
    [button addTarget:self action:@selector(didTapButton:) forControlEvents:UIControlEventTouchUpInside];
    [self.contentView setUserInteractionEnabled:YES];
    [self.contentView setMultipleTouchEnabled:YES];

    
    [NSLayoutConstraint activateConstraints:@[
       [self.textView.topAnchor constraintEqualToAnchor:self.contentView.topAnchor],
       [self.textView.leadingAnchor constraintEqualToAnchor:self.contentView.leadingAnchor],
       [self.textView.trailingAnchor constraintEqualToAnchor:self.contentView.trailingAnchor],
       
       [self.processConfigButton.topAnchor constraintEqualToAnchor:self.textView.bottomAnchor],
       [self.processConfigButton.leadingAnchor constraintEqualToAnchor:self.textView.leadingAnchor],
       [self.processConfigButton.trailingAnchor constraintEqualToAnchor:self.textView.trailingAnchor],
       [self.processConfigButton.bottomAnchor constraintEqualToAnchor:self.contentView.bottomAnchor]
    ]];
}

- (void)prepareForReuse {
    [super prepareForReuse];
    self.textView.text = nil;
}

- (void)didTapButton:(id)sender {
}


@end

File: Branch-SDK/NativeComputeDebugButtonCell.swift
import UIKit


@objc class NativeComputeDebugButtonCell: UITableViewCell {

    weak var textView: UITextView!
    weak var processConfigUrlsButton: UIButton!
    
    var buttonTapCallback: () -> ()  = { }
    
    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        self.initialize()
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        self.initialize()
    }
    
    @objc func didTapButton() {
        buttonTapCallback()
    }

    func initialize() {
        // textview config
        let textView = UITextView(frame: .zero)
        textView.translatesAutoresizingMaskIntoConstraints = false
        textView.isEditable = false
        textView.isScrollEnabled = false
        textView.textAlignment = .left
        textView.backgroundColor = .lightGray
        textView.textColor = .black
        self.contentView.addSubview(textView)
        self.textView = textView
        
        // button config
        let button = UIButton(type: .roundedRect)
        button.translatesAutoresizingMaskIntoConstraints = false
        button.setTitle("Process All Config URLs", for: .normal)
        button.backgroundColor = .black
        button.setTitleColor(.lightGray, for: .normal)
        button.titleLabel?.font = UIFont.systemFont(ofSize: 14)
        button.isEnabled = true
        button.isUserInteractionEnabled = true
        self.contentView.addSubview(button)
        self.processConfigUrlsButton = button
        self.processConfigUrlsButton.addTarget(self, action: #selector(didTapButton), for: .touchUpInside)
        self.contentView.isUserInteractionEnabled = true
        self.contentView.isMultipleTouchEnabled = true


        NSLayoutConstraint.activate([
            self.textView.topAnchor.constraint(equalTo: self.contentView.topAnchor),
            self.textView.leadingAnchor.constraint(equalTo: self.contentView.leadingAnchor),
            self.textView.trailingAnchor.constraint(equalTo: self.contentView.trailingAnchor),
            
            self.processConfigUrlsButton.topAnchor.constraint(equalTo: self.textView.bottomAnchor),
            self.processConfigUrlsButton.leadingAnchor.constraint(equalTo: self.contentView.leadingAnchor),
            self.processConfigUrlsButton.trailingAnchor.constraint(equalTo: self.contentView.trailingAnchor),
            self.processConfigUrlsButton.bottomAnchor.constraint(equalTo: self.contentView.bottomAnchor)
        ])
    }

    override func prepareForReuse() {
        super.prepareForReuse()
        self.textView.text = nil
    }
}
